File: 121. Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if(prices.size()<2)
            return 0;
        
        int lowest = prices[0], maxProfit = 0;
        for(int i=1; i<prices.size(); i++) {
            if(prices[i]<lowest)
                lowest = prices[i];
            else
                maxProfit = max(maxProfit, prices[i]-lowest);
        }
        
        return maxProfit;
    }
};

File: 121. Best Time to Buy and Sell Stock.java
class Solution {
    // Time: O(N), Space: O(1)
    public int maxProfit(int[] prices) {
        int lowestPrice = prices[0], maxProfit = 0;
        for(int price: prices) {
            maxProfit = Math.max(maxProfit, price - lowestPrice);
            lowestPrice = Math.min(lowestPrice, price);
        }

        return maxProfit;
    }
}
